multidict/_multidict_py.py

```diff
@@ -160,6 +169,38 @@ def __repr__(self) -> str:
         body = ", ".join(lst)
         return "<{}({})>".format(self.__class__.__name__, body)
 
+    def __and__(self, other: Iterable[str]) -> Union[set[str], NotImplementedType]:
+        ret = set()
+        try:
+            it = iter(other)
+        except TypeError:
+            return NotImplemented
+        for key in it:
+            identity = self._keyfunc(key)
+            for i, k, v in self._impl._items:
+                if i == identity:
+                    ret.add(key)
+        return ret
+
+    def __or__(self, other: Iterable[_T]) -> Union[set[Union[str, _T]], NotImplementedType]:
+        ret: set[Union[str, _T]] = set(self)
+        try:
+            it = iter(other)
+        except TypeError:
+            return NotImplemented
+        for key in it:
+            if not isinstance(key, str):
+                ret.add(key)
+                continue
```

Comment:
As the runtime behaviour was incorrect in this case, probably want another test to verify any C code works correctly too.

Reply:
tests are added